src/Microsoft.AspNetCore.Rewrite/RewriteContext.cs

```diff
@@ -34,6 +34,11 @@ public class RewriteContext
         /// </summary>
         public RuleResult Result { get; set; }
 
+        /// <summary>
+        /// Represents whether or not the rule is a global rule
+        /// </summary>
+        public bool GlobalRule { get; set; }
```

Comment:
Let's try to remove this as it's specific to only one rule system. it's only used in UrlSegment? Maybe we need a separate GlobalRuleUrlSegment instead.